optimade/models/entries.py

```diff
@@ -147,12 +177,48 @@ class EntryResource(Resource):
 Database-provider-specific properties need to include the database-provider-specific prefix (see section on Database-Provider-Specific Namespace Prefixes).""",
     )
 
+    meta: Optional[EntryMetadata] = StrictField(
+        None,
+        description="""A dictionary, containing entry and property-specific metadata for a given entry.""",
+    )
+
     relationships: Optional[EntryRelationships] = StrictField(
         None,
         description="""A dictionary containing references to other entries according to the description in section Relationships encoded as [JSON API Relationships](https://jsonapi.org/format/1.0/#document-resource-object-relationships).
 The OPTIONAL human-readable description of the relationship MAY be provided in the `description` field inside the `meta` dictionary of the JSON API resource identifier object.""",
     )
 
+    @root_validator(pre=True)
+    def check_meta(cls, values):
+        """Validator to check whether meta field has been formatted correctly."""
+        from optimade.server.mappers.entries import (
+            BaseResourceMapper,
+        )
+
+        meta = values.get("meta")
+        if not meta:
+            return values
+
+        if property_metadata := meta.pop("property_metadata", None):
+            # check that all the fields under property metadata are in attributes
+            attributes = values.get("attributes", {})
+            for subfield in property_metadata:
+                if subfield not in attributes:
+                    raise ValueError(
+                        f"The keys under the field `property_metadata` need to match with the field names in attributes. The field {subfield} is however not in attributes."
+                    )
+
+        # At this point I am getting ahead of the specification. There is the intention to allow database specific fields(with the database specific prefixes) here in line with the JSON API specification, but it has not been decided yet how this case should be handled in the property definitions.
```

Comment:
This makes sense, thanks. Rickard and I were discussing this recently -- perhaps in our config we should allow a list of prefixes already that we can check here?

Reply:
I have added a field to the config where you can list the supported fields.